pkgs/os-specific/linux/msi-ec/default.nix

```diff
@@ -0,0 +1,47 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  linuxPackages,
+  kernel ? linuxPackages.kernel,
+}:
+stdenv.mkDerivation {
+  pname = "msi-ec-kmods";
+  version = "0-unstable-2024-09-19";
+
+  src = fetchFromGitHub {
+    owner = "BeardOverflow";
+    repo = "msi-ec";
+    rev = "94c2a45c04a07096e10d7cb1240e1a201a025dc0";
+    hash = "sha256-amJUoIf5Sl62BLyHLeam2fzN1s+APoWh2dH5QVfJhCs=";
+  };
+
+  dontMakeSourcesWritable = false;
+
+  postPatch = ''
+    cp ${./patches/Makefile} ./Makefile
+  '';
+
+  hardeningDisable = [ "pic" ];
+
+  makeFlags = kernel.makeFlags ++ [
+    "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+    "ccflags-y+=-Wno-error=incompatible-pointer-types"
```

Comment:
```suggestion

```
This should be unneeded since:
https://github.com/BeardOverflow/msi-ec/commit/a42a91536852c5508d6dd62b2172c55ee9d7edf8